assets/src/js/Components/ArchivePicker/ArchivePicker.js

```diff
@@ -5,10 +5,15 @@ import ArchivePickerList from './ArchivePickerList';
 import SingleSidebar from './SingleSidebar';
 import MultiSidebar from './MultiSidebar';
 import ArchivePickerToolbar from './ArchivePickerToolbar';
+import {updateImageBlockAttributes, updateHappyPointAttributes} from './blockUpdateFunctions';
 
 const {apiFetch, url, i18n} = wp;
 const {addQueryArgs} = url;
 const {__} = i18n;
+const timeout = delay => {
```

Comment:
@Osong-Michael I'm not sure if I understand why we need to implement this `timeout` 🙏

Reply:
So for one, when requesting image details back from `WP` when the images are first added to the Media Library, `WP` will sometime return undefined even though the image is in the Media Library, some resources I found online tell you to run the function in `async` mode with a `timeout` so the details can be fetched, the best way I found to execute this is run the fetch the first time, if we get undefined, wait a few seconds then run it again, I also made the timeout reusable so i can use it in a few different places too.

Reply:
Oh I see. It's like a `sleep()`, am I correct?